### ISSUE TITLE
Build error with @embroider/macros and static helpers

### ISSUE THREAD

**Issue by simonihmig** [embroider-build/embroider#476]
When adding `staticHelpers: true` to the embroider config for `ember-bootstrap` (which uses `@embroider/macros` as a v1 addon), as suggested in the addon guide, I get this build failure:

```
Error: Missing helper macroCondition in /private/var/folders/q3/yzggf_l965zbzps_bk8xr37c0000gn/T/embroider/9d4dc2/templates/components/bs-accordion.hbs
    at CompatResolver.dependenciesOf (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/compat/src/resolver.js:184:31)
    at TemplateCompiler.precompile (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/core/src/template-compiler.js:246:49)
    at TemplateCompiler.compile (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/core/src/template-compiler.js:255:47)
    at Object.hbsLoader (/Users/simonihmig/Projects/ember-bootstrap/node_modules/@embroider/webpack/src/webpack-hbs-loader.js:10:16)
```

See [this PR](https://github.com/kaliber5/ember-bootstrap/pull/1125) and its [build error](https://github.com/kaliber5/ember-bootstrap/pull/1125/checks?check_run_id=778459435#step:5:69)

**Comment by ef4** [comment]
Probably we need to make sure that the ast transform that does component & helper resolution runs after the ast transform that implements the macros.

**Comment by simonihmig** [comment]
@ef4 I tried to take a stab at this today. Tbh it was a bit overwhelming working at this codebase, with my so far limited understanding how everything works together. So I certainly need a bit of guidance here.

*However* I made some progress, so these are my findings so far:

It seems the macro AST is intentionally excluded from running during stage 1 here: https://github.com/embroider-build/embroider/blob/master/packages/compat/src/v1-addon.ts#L106. Which makes sense, as v2 addons are supposed to still use the macros and not have them compiled away, to my understanding at least.

I added the macros to `builtInHelpers` in https://github.com/embroider-build/embroider/blob/master/packages/compat/src/resolver.ts#L44-L81. This seems to prevent the template compiler / resolver to choke on them. And indeed the build succeeded, and the test suite (of ember-bootstrap) ran and almost passed (just one failing test, which I did not investigate, but is certainly not related to the same issue). It would have totally failed if anything related to the macros would not have worked, so that's a pretty good sign!

I wonder if that's also (architecturally) the "right" way to do it, or some other approach is needed?

Happy to work on a PR if the direction is clear!
